src/Enzo/enzo_EnzoEOSIdeal.cpp

```diff
@@ -0,0 +1,409 @@
+			  std::string func_name)
+{
+  for (std::size_t i = 0; i<group_names.size(); i++){
+    std::string group_name = group_names[i];
+    int num_ref_fields = grouping_ref.size(group_name);
+    int num_check_fields = grouping_check.size(group_name);
+
+    ASSERT3(func_name.c_str(),
+	    ("%s and %s groupings must have the same number of entries "
+	     "for the %s group"),
+	    ref_name.c_str(), check_name.c_str(), group_name.c_str(),
+	    num_ref_fields == num_check_fields);
+
+    for (int j=0;j<num_ref_fields;j++){
+      std::string ref_field = grouping_ref.item(group_name,j);
+      std::string check_field = grouping_check.item(group_name,j);
+
+      ASSERT4(func_name.c_str(),
+	      ("Field %d of the %s group is expected to have the same name in "
+	       "the %s and %s groupings"),
+	      j, group_name.c_str(), ref_name.c_str(), check_name.c_str(),
+	      ref_field == check_field);
+    }
+  }
+}
+
+//----------------------------------------------------------------------
+
+void check_recon_integ_overlap_(Grouping &reconstructable_group,
+				Grouping &integrable_group,
+				std::string func_name)
+{
+  // We assume that the following groups are represented by the same fields in
+  // integrable and reconstructable
+  std::vector<std::string> common_groups = {"density", "velocity", "bfield"};
+
+  // we also expect overlap with the passive scalars
+  std::vector<std::string> scalar_groups =
+    EnzoCenteredFieldRegistry::passive_scalar_group_names();
+  common_groups.insert(common_groups.end(), scalar_groups.begin(),
+		       scalar_groups.end());
+    
+  confirm_same_fields_(integrable_group, reconstructable_group,
+		       "integrable", "reconstructable", common_groups,
+		       func_name);
+}
+
+//----------------------------------------------------------------------
+
+void EnzoEOSIdeal::reconstructable_from_integrable
+  (Block *block, Grouping &integrable_group, Grouping &reconstructable_group,
+   Grouping &conserved_passive_group, int stale_depth) const
+{
+
+  // Confirm that the expected fields (e.g. density, vx, vy, vz, bx, by, bz)
+  // are the same in reconstructable_group and integrable_group
+  check_recon_integ_overlap_(reconstructable_group, integrable_group,
+			     "EnzoEOSIdeal::reconstructable_from_integrable");
+
+  // Simply compute the pressure
+  std::string pressure_name = reconstructable_group.item("pressure",0);
+  pressure_from_integrable(block, integrable_group, pressure_name,
+			   conserved_passive_group, stale_depth);
+}
+
+//----------------------------------------------------------------------
+
+void EnzoEOSIdeal::integrable_from_reconstructable
+  (Block *block, Grouping &reconstructable_group, Grouping &integrable_group,
+   int stale_depth, int reconstructed_axis) const
+{
+
+  if (grackle_variable_gamma_()){
+    // we would need to model a field with the
+    ERROR("EnzoEOSIdeal::pressure_from_integrable",
+	  "Doesn't currently support spatial variations in gamma");
+  }
+
```

Comment:
```suggestion

#ifdef CONFIG_USE_GRACKLE
  if (grackle_variable_gamma_()){
    // we would need to model a field with the
    ERROR("EnzoEOSIdeal::pressure_from_integrable",
	  "Doesn't currently support spatial variations in gamma");
  }
#endif

```

Reply:
This actually change actually isn't correct. There is limited support for Grackle - just not when the adiabatic index is allowed to vary

Reply:
Nevermind. I missed that `grackle_variable_gamma_` is an internal function that includes the corresponding `#ifdefs` and thought that if the grackle machinery is not compiled in that this check may lead to erros.

Reply:
I basically though this would be sth like https://github.com/enzo-project/enzo-e/pull/58/files